common/src/auth/SharedAccessSignature.cs

```diff
@@ -0,0 +1,130 @@
+            {
+                throw new FormatException(string.Format(CultureInfo.InvariantCulture, "Missing field: {0}", SharedAccessSignatureConstants.SignatureFieldName));
+            }
+
+            string expiry;
+            if (!parsedFields.TryGetValue(SharedAccessSignatureConstants.ExpiryFieldName, out expiry))
+            {
+                throw new FormatException(string.Format(CultureInfo.InvariantCulture, "Missing field: {0}", SharedAccessSignatureConstants.ExpiryFieldName));
+            }
+
+            // KeyName (skn) is optional .
+            string keyName;
+            parsedFields.TryGetValue(SharedAccessSignatureConstants.KeyNameFieldName, out keyName);
+
+            string encodedAudience;
+            if (!parsedFields.TryGetValue(SharedAccessSignatureConstants.AudienceFieldName, out encodedAudience))
+            {
+                throw new FormatException(string.Format(CultureInfo.InvariantCulture, "Missing field: {0}", SharedAccessSignatureConstants.AudienceFieldName));
+            }
+
+            return new SharedAccessSignature(
+                shareAccessSignatureName,
+                SharedAccessSignatureConstants.EpochTime + TimeSpan.FromSeconds(double.Parse(expiry, CultureInfo.InvariantCulture)),
+                expiry, keyName, signature, encodedAudience);
+        }
+
+        public static bool IsSharedAccessSignature(string rawSignature)
+        {
+            if (string.IsNullOrWhiteSpace(rawSignature))
+            {
+                return false;
+            }
+
+            IDictionary<string, string> parsedFields = ExtractFieldValues(rawSignature);
+            string signature;
+            bool isSharedAccessSignature = parsedFields.TryGetValue(SharedAccessSignatureConstants.SignatureFieldName, out signature);
+
+            return isSharedAccessSignature;
+        }
+
+        public bool IsExpired()
+        {
+            return ExpiresOnUtc + SharedAccessSignatureConstants.MaxClockSkew < DateTime.UtcNow;
+        }
+
+        protected static IDictionary<string, string> ExtractFieldValues(string sharedAccessSignature)
+        {
+            string[] lines = sharedAccessSignature.Split();
+
+            if (!string.Equals(lines[0].Trim(), SharedAccessSignatureConstants.SharedAccessSignature, StringComparison.Ordinal) || lines.Length != 2)
+            {
+                throw new FormatException("Malformed signature");
+            }
+
+            IDictionary<string, string> parsedFields = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            string[] fields = lines[1].Trim().Split(new string[] { SharedAccessSignatureConstants.PairSeparator }, StringSplitOptions.None);
+
+            foreach (string field in fields)
+            {
+                if (!string.IsNullOrEmpty(field))
+                {
+                    string[] fieldParts = field.Split(new string[]{ SharedAccessSignatureConstants.KeyValueSeparator }, StringSplitOptions.None);
```

Comment:
This is creating an array of that one const char on each field in fields. Consider adding the array to SharedAccessSignatureConstants, or if you prefer as a static readonly on this class. You'll reduce unnecessary memory allocations.